pkg/kosmosctl/unjoin/unjoin.go

```diff
@@ -108,33 +108,23 @@ func (o *CommandUnJoinOptions) Validate() error {
 
 func (o *CommandUnJoinOptions) Run(f ctlutil.Factory, cmd *cobra.Command, args []string) error {
 	//delete cluster
-	err := o.DynamicClient.Resource(schema.GroupVersionResource{
-		Group:    "kosmos.io",
-		Version:  "v1alpha1",
-		Resource: "clusters",
-	}).Namespace("").Delete(context.TODO(), args[0], metav1.DeleteOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("(cluster) kosmosctl unjoin run error, delete cluster failed: %s", err)
-	}
-
 	for {
-		// Wait 3 second
-		time.Sleep(3 * time.Second)
-		_, err := o.DynamicClient.Resource(schema.GroupVersionResource{
+		err := o.DynamicClient.Resource(schema.GroupVersionResource{
 			Group:    "kosmos.io",
 			Version:  "v1alpha1",
 			Resource: "clusters",
-		}).Namespace("").Get(context.TODO(), args[0], metav1.GetOptions{})
+		}).Namespace("").Delete(context.TODO(), args[0], metav1.DeleteOptions{})
```

Comment:
here logic seems be changed, and the part that originally checked if the cluster was deleted has been removed, why?

Reply:
Simplified code, delete instead of get
now logic: first delete -> wait -> next delete 
if delete success firstly, 'next delete' will return not found error. we will break.